### PR TITLE
HOLD: expanded test suite proving writeQuery triggers observable update

### PR DESCRIPTION
**Pull request by rosskevin**
---
HOLD: this is based on big PR #4261.  Once it is merged and this is rebased, the change will be very small and only to `packages/apollo-client/src/__tests__/ApolloClient.ts`
---

### Checklist:

- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

The impetus for expanding the test suite is https://github.com/apollographql/apollo-client/issues/2415#issuecomment-449757616

By expanding this test suite, it proves without a doubt that `mutate` with standard payloads as well as `writeQuery` trigger observable updates.  apollo-client is working as expected and this just rules out standard usage as a bug that some are seeing in #2415.

### CODE REVIEW

File: packages/apollo-client/src/__tests__/ApolloClient.ts [hunk 1]
```diff
@@ -1129,46 +1147,227 @@ describe('ApolloClient', () => {
...
+                // this should re call next
+                client.writeQuery({
+                  query,
+                  data: {
+                    friends: bestFriends,
+                    __typename: 'Person',
+                  },
+                });
+
+                setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (danilobuerger) [reply packages/apollo-client/src/__tests__/ApolloClient.ts]: Your test is incorrect. Try `client.writeQuery<Data>`. Format of `data` should be:
```
     client.writeQuery<Data>({
      query,
      data: {
        people: {
          id: 1,
          friends: bestFriends,
          __typename: 'Person',
        }
      },
    });
```

[the remark this answers is not in the document]
Comment (rosskevin) [reply packages/apollo-client/src/__tests__/ApolloClient.ts]: So the funniest part about this is that I did all the work on the base PR to get type checking in tests, and I omitted the parameterized type which would have exposed my error!

**Comment by hwillson** [comment]
@rosskevin Would you be able to submit a new PR (and close this one) that just has the `packages/apollo-client/src/__tests__/ApolloClient.ts` changes? It sounds like those changes can be merged in without the changes in #4261, so we can get that done sooner than later. Thanks!

**Comment by rosskevin** [comment]
Closed in favor of cherrypicked changes in #4281

**Pull request closed without merging**